### DISCUSSION TITLE
Login fails on Android with Firefox

### DISCUSSION THREAD

**Discussion by henrebotha** [community/community#65753] [Mobile]
### Select Topic Area

Bug

### Body

GitHub 1.128.1, Firefox 116.3.0, Samsung S10 on Android 12. I have 2FA enabled with the app. When I try to log in to GitHub in Firefox, and the "open links in external apps" setting is set to "ask every time", I can't log in. If I allow the app to open when prompted, the app says "something went wrong". If I disallow it, I get a redirect loop back to the login form. Only when I set Firefox to always open in external apps does the login flow work.

This definitely sounds like it's related to the recent issues with the Brave browser.

**Comment by alcere** [comment]
Hi @henrebotha, thanks for reporting this issue! We are currently investigating the root cause of the problem, meanwhile, could you give it a try again using our beta release? We have introduced some changes in the sig-in flow that could help here. Thanks in advanced!

**Reply by henrebotha** [reply]
@alcere I tried the beta and it worked as expected. However, a small wrinkle now: It seems that visiting GitHub links in Firefox no longer prompts me to open them in the app. I can force the issue (Firefox has an explicit button for "open in app"), but it's surprising to me that clicking a GitHub link now just stays in the browser by default. Worse, I can't tell if I prefer this behaviour. 😅 Something for your UX folks to ponder, I'm sure.

**Reply by alcere** [reply]
Thanks @henrebotha for trying! 
> However, a small wrinkle now: It seems that visiting GitHub links in Firefox no longer prompts me to open them in the app. I can force the issue (Firefox has an explicit button for "open in app"), but it's surprising to me that clicking a GitHub link now just stays in the browser by default

This is very interesting! Can you give us some more details about this? For instance, in the flow you are describing: do you manually introduce the GitHub URL in the browser? or are you trying to open a GitHub link, i.e: from an email in the browser?

**Reply by henrebotha** [reply]
New flow:
1. Set Firefox "open links in apps" setting to "ask before opening" (or "always").
2. Visit github.com and click any link, such as a repo.

I expect to get a prompt from Firefox to open the link in the app (or for it to simply open the link in the app without prompting), but it doesn't happen.

**Reply by alcere** [reply]
I appreciate you sharing more details, unfortunately this ask is not something GitHub would be able to do as that is something Forefox would have to enable. It looks like they have a community similar to ours where you can share feedback: https://connect.mozilla.org/. Best of luck with your request!

**Comment by christianborden2000** [comment]
> ### Select Topic Area
> 
> Bug
> 
> ### Body
> 
> GitHub 1.128.1, Firefox 116.3.0, Samsung S10 on Android 12. I have 2FA enabled with the app. When I try to log in to GitHub in Firefox, and the "open links in external apps" setting is set to "ask every time", I can't log in. If I allow the app to open when prompted, the app says "something went wrong". If I disallow it, I get a redirect loop back to the login form. Only when I set Firefox to always open in external apps does the login flow work.
> 
> This definitely sounds like it's related to the recent issues with the Brave browser.
